README.md

```diff
@@ -1,113 +1,323 @@
+    }
+
+    constructor(string memory symbol_, string memory name_, string memory version_, uint256 chainId_) public {
+        symbol = symbol_;
+        name   = name_;
+        totalSupply = _totalSupply;
+        balances[msg.sender] = _totalSupply;
+    }
+
+    // --- Token ---
+    function transfer(address to, uint value) public returns (bool) {
+        transferFrom(msg.sender, to, value);
+        return true;
+    }
+    function transferFrom(address from, address to, uint value)
+        public returns (bool)
+    {
+        if (from != msg.sender && allowance[from][msg.sender] != uint(-1)) {
+            allowance[from][msg.sender] = sub(allowance[from][msg.sender], value);
+        }
+        balanceOf[from] = sub(balanceOf[from], value);
+        balanceOf[to] = add(balanceOf[to], value);
+        emit Transfer(from, to, value);
+        return true;
+    }
+    function approve(address spender, uint value) public returns (bool) {
+        allowance[msg.sender][spender] = value;
+        emit Approval(msg.sender, spender, value);
+        return true;
+    }
+}
+```
+
+The behaviours of the constructor and the functions `transfer` and
+`transferFrom` can be specified as the following Act:
+
+```act
+behaviour init of Token
+interface constructor(string _symbol, string _name, string _version, uint _totalSupply)
+
+creates
+    string name         := _name
+    string symbol       := _symbol
+    uint256 totalSupply := _totalSupply
+    mapping(address => uint) balanceOf :=  [CALLER := _totalSupply]
+    mapping(address=>mapping(address=>uint)) allowance := []
+
+
+behaviour transfer of Token
+interface transfer(uint256 value, address to)
+
+iff
+
+   CALLVALUE == 0
+   value <= balanceOf[CALLER]
+   CALLER =/= to => balanceOf[to] + value < 2^256
+
+case CALLER =/= to:
+
+   storage
+
+     balanceOf[CALLER] => balanceOf[CALLER] - value
+     balanceOf[to]     => balanceOf[to] + value
+
+   returns 1
+
+case CALLER == to:
+
+   returns 1
+
+behaviour transferFrom of Token
+interface transferFrom(address src, address dst, uint amount)
+
+iff
+
+   value <= balanceOf[CALLER]
+   src    =/= dst => balanceOf[dst] + value < 2^256
+   CALLER =/= src => 0 <= allowance[src][CALLER] - value
+   CALLVALUE == 0
+
+case src =/= dst:
+
+   case CALLER == src:
+
+      storage
+
+         balances[src] => balances[src] - value
+         balances[dst] => balances[dst] + value
+
+      returns 1
+
+   case CALLER =/= src and allowance[src][CALLER] == 2^256 - 1:
+
+      storage
+
+         balances[src] => balances[src] - value
+         balances[dst] => balances[dst] + value
+
+      returns 1
+
+   case _:
+
+      storage
+
+         allowance[src][CALLER] => allowance[src][CALLER] - value
+         balances[src]          => balances[src] - value
+         balances[dst]          => balances[dst] + value
+
+      returns 1
+
+case src == dst:
+
+   returns 1
 ```
 
+Parsing te Act gives us the generated proof obligations:
+
+```JSON
```

Comment:
notice that this json is from a simpler, `mint` spec and not the `transfer`  specified above.

Reply:
Oh. Do we have the spec somewhere? Or the json for the transfer above?

Reply:
Not currently, but it's probably the next reachable target

Reply:
Should I update the act and code to `mint`, or just remove the json?

Reply:
It's good to show the json I think, so yeah let's show the spec of mint right now